File: apps/api/app/models/meal.py
import uuid
from datetime import datetime
from decimal import Decimal

import sqlalchemy as sa
from sqlalchemy import DateTime, ForeignKey, Numeric, String
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base

MealSource = sa.Enum("fatsecret", "custom", "seed", name="meal_source", create_type=False)
# V10: 한글값 → 영문 키. 라벨 i18n 은 클라이언트 책임.
MealFoodKind = sa.Enum(
    "kibble", "treat", "regular", "prescription",
    name="meal_food_kind",
    create_type=False,
)


class Meal(Base):
    __tablename__ = "meal"

    id: Mapped[uuid.UUID] = mapped_column(primary_key=True, default=uuid.uuid4)
    pet_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey("pet.id", ondelete="CASCADE"), nullable=False
    )
    ts: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    source: Mapped[str] = mapped_column(MealSource, nullable=False)
    food_id: Mapped[str | None] = mapped_column(String, nullable=True)
    food_name: Mapped[str] = mapped_column(String, nullable=False)
    # V10: Float → Numeric(10, 2) — 영양정보 정밀도.
    qty_g: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    kcal: Mapped[Decimal | None] = mapped_column(Numeric(10, 2), nullable=True)
    protein_g: Mapped[Decimal | None] = mapped_column(Numeric(10, 2), nullable=True)
    carbs_g: Mapped[Decimal | None] = mapped_column(Numeric(10, 2), nullable=True)
    fat_g: Mapped[Decimal | None] = mapped_column(Numeric(10, 2), nullable=True)
    note: Mapped[str | None] = mapped_column(String, nullable=True)
    food_kind: Mapped[str | None] = mapped_column(MealFoodKind, nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=sa.func.now()
    )

File: apps/api-java/src/main/java/com/petfinect/api/domain/Meal.java
package com.petfinect.api.domain;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/** 식사 기록 (W2). pet 별 시계열. */
@Entity
@Table(name = "meal", indexes = @Index(name = "ix_meal_pet_ts", columnList = "pet_id,ts"))
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Meal {

	@Id
	@GeneratedValue(strategy = GenerationType.UUID)
	private UUID id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "pet_id", nullable = false)
	private Pet pet;

	@Column(nullable = false)
	private OffsetDateTime ts;

	@Enumerated(EnumType.STRING)
	@JdbcTypeCode(SqlTypes.NAMED_ENUM)
	@Column(name = "source", nullable = false, columnDefinition = "meal_source")
	private MealSource source;

	@Column(name = "food_id")
	private String foodId;

	@Column(name = "food_name", nullable = false)
	private String foodName;

	// V10: DOUBLE PRECISION → NUMERIC(10, 2) — 영양정보 정밀도 (BigDecimal).
	@Column(name = "qty_g", nullable = false)
	private BigDecimal qtyG;

	private BigDecimal kcal;

	@Column(name = "protein_g")
	private BigDecimal proteinG;

	@Column(name = "carbs_g")
	private BigDecimal carbsG;

	@Column(name = "fat_g")
	private BigDecimal fatG;

	private String note;

	@Enumerated(EnumType.STRING)
	@JdbcTypeCode(SqlTypes.NAMED_ENUM)
	@Column(name = "food_kind", columnDefinition = "meal_food_kind")
	private MealFoodKind foodKind;

	@CreationTimestamp
	@Column(name = "created_at", nullable = false, updatable = false)
	private OffsetDateTime createdAt;
}
